applications/cryptex.md

```diff
@@ -0,0 +1,297 @@
+
+### Project Details
+
+The major pieces:
+1. [IBS Block Seal](#ibe-block-seal-in-aura)
+2. [Encryption-to-Future](#encryption-to-future-slots)
+
+For a more in-depth (informal) look at the mathematics behind this, look at [etf_aura.pdf](https://github.com/ideal-lab5/Grants-Program/blob/ibe/applications/etf_aura.pdf)
+
+#### What this is not
+- this does not use a proof of stake consensus. For the scope of the proposal, we are assuming a static, well defined validator set using PoA consensus based on Aura. 
+- the proposal does not highlight any specific privacy preserving tools nor does it use threshold signatures or any specific MPC protocols 
+- most of that is scoped for the future though
+
+#### Notation and Terminology
+
+For the following, assume that we are using curve BLS12-381. As such, we will refer to its scalar field using $\mathbb{Z}_p$. Similarly, we have the pairing friendly elliptic curve groups $\mathbb{G}_1$ and $\mathbb{G}_2$.
+
+#### Background
+
+**Threshold Secret Sharing**
+  Threshold secret sharing (TSS) was first introduced by [Shamir in 1979](https://web.mit.edu/6.857/OldStuff/Fall03/ref/Shamir-HowToShareASecret.pdf). Our protocol uses a TSS scheme with a semi-trusted coordinator to obtain (*not derive) a shared session secret key and shared session public key among the session validator set. We explain this below, and more in depth in [etf_aura.pdf](https://github.com/ideal-lab5/Grants-Program/blob/ibe/applications/etf_aura.pdf).
+  
+**Identity Based Cryptography**
+  Identity Based Cryptography (IBC) was first introduced by [Shamir in 1984](https://link.springer.com/content/pdf/10.1007/3-540-39568-7_5.pdf). Since then, there has been significant research into identity based cryptosystems, I present an overview to the reader [here](https://csrc.nist.gov/csrc/media/Presentations/2023/stppa5-ibe/images-media/20230209-stppa5-Dan-Boneh--IBE.pdf). A complete IBC consists of six algorithms: `Setup, Keygen, Encrypt, Decrypt, Sign, Verify`. Our network will leverage the sign and verify functionalities to seal and verify blocks, and will use the encrypt and decrypt functionalities to enable encryption-to-future-slots.
+
+  We will use the [Boneh-Franklin identity based encryption scheme](https://crypto.stanford.edu/~dabo/papers/bfibe.pdf) (BF IBE) in this proposal in order to encrypt messages to future slots. 
+
+#### IBS Block Seal in Aura
+
+The goal of this is to introduce an IBS-BLS block seal in Aura, where each slot has a unique role associated with it that we can encrypt to. In the future, we intend to migrate to [Sassafras](https://eprint.iacr.org/2023/031.pdf). This is the first step to building encryption to the future.
+
+https://docs.rs/sc-consensus-aura/latest/sc_consensus_aura/
+
+We will create a fork of Aura, wherein blocks are sealed using identity based signatures, where each slot has a role associated with it.
```

Comment:
An IBS is something else.  This is just a DLEQ proof of a BLS signature on the slot seed, computed similarly to Babe.  The DLEQ proof contains a Fiat-Shamir transform, which can sign the block, just like Schnorr.

It's possible to have a separate Schnorr signature which signs both the BLS signature and the block, but checking a BLS signature is slower than checking a DLEQ proof of a BLS signature, so not much point.